### PR TITLE
Add Makefile to automate compilation step

### PR DESCRIPTION
**Pull request by dimroc**
Can now run `make` in top level repo folder to simplify compilation for issue https://github.com/smartcontractkit/chainlink/issues/98

### CODE REVIEW

**Comment by thodges-gh** [comment]
Tested on a fresh VM. Still leaving some assumptions in the Install heading like working with the correct directories in order for the instructions to remain simple.

File: README.md [hunk 1]
```diff
@@ -14,8 +14,10 @@ Examples of how to utilize and integrate Chainlinks can be found in our [Hello C
 ## Install
 
 1. [Install Go 1.9+](https://golang.org/doc/install#install), and add your GOPATH's [bin directory to your PATH](https://golang.org/doc/code.html#GOPATH)
-2. Install ChainLink: `$ go get -u github.com/smartcontractkit/chainlink`
-3. Run the node: `$ chainlink help`
+2. Install [dep](https://github.com/golang/dep#installation): `$ brew install dep` <br> or `$ go get -u github.com/golang/dep/cmd/dep`
+3. Clone ChainLink: `$ git clone github.com/smartcontractkit/chainlink`
```
Comment (se3000) [inline README.md]: Trying this out locally, the URL needs "https://" in front of it:

3. Clone ChainLink: `$ git clone https://github.com/smartcontractkit/chainlink`

Comment (thodges-gh) [reply README.md]: Ah, right. Got that updated in both places for cloning.

**Comment by se3000** [comment]
Looks great, small change to the instructions.

**Comment by se3000** [comment]
@thodges-gh those assumptions work for me, at least for now. Thanks!

**Comment by recursiev** [comment]
Not sure if I've done something wrong, but following the new instructions:
- New VM (Ubuntu 16.04)
- Extract Go 1.9.4 to /usr/local/go, update PATH
$ go get -u github.com/golang/dep/cmd/dep
$ git clone https://github.com/smartcontractkit/chainlink.git
^^ have to move chainlink to /go/src
$ cd /go/src/chainlink
$ make install
$ chainlink node
(geth already running, listening on 127.0.0.1:8546 and ETH_URL set to ws://localhost:8546)
Then a new error message:
{"level":"info","ts":1519337632.3949947,"caller":"logger/logger.go:70","msg":"Starting Chainlink Node 0.2.0 at commit unset"}
{"level":"fatal","ts":1519337632.40284,"caller":"logger/logger.go:113","msg":"[bad status]","stacktrace":"chainlink/vendor/github.com/smartcontractkit/chainlink/logger.Fatal
	/home/ubuntu/go/src/chainlink/vendor/github.com/smartcontractkit/chainlink/logger/logger.go:113
chainlink/vendor/github.com/smartcontractkit/chainlink/store.NewStore
	/home/ubuntu/go/src/chainlink/vendor/github.com/smartcontractkit/chainlink/store/store.go:37
chainlink/vendor/github.com/smartcontractkit/chainlink/services.NewApplication
	/home/ubuntu/go/src/chainlink/vendor/github.com/smartcontractkit/chainlink/services/application.go:31
chainlink/vendor/github.com/smartcontractkit/chainlink/cmd.ChainlinkAppFactory.NewApplication
	/home/ubuntu/go/src/chainlink/vendor/github.com/smartcontractkit/chainlink/cmd/client.go:121
chainlink/vendor/github.com/smartcontractkit/chainlink/cmd.(*ChainlinkAppFactory).NewApplication
	<autogenerated>:1
chainlink/vendor/github.com/smartcontractkit/chainlink/cmd.(*Client).RunNode
	/home/ubuntu/go/src/chainlink/vendor/github.com/smartcontractkit/chainlink/cmd/client.go:37
chainlink/vendor/github.com/smartcontractkit/chainlink/cmd.(*Client).RunNode-fm
	/home/ubuntu/go/src/chainlink/main.go:46
chainlink/vendor/github.com/urfave/cli.HandleAction
	/home/ubuntu/go/src/chainlink/vendor/github.com/urfave/cli/app.go:490
chainlink/vendor/github.com/urfave/cli.Command.Run
	/home/ubuntu/go/src/chainlink/vendor/github.com/urfave/cli/command.go:210
chainlink/vendor/github.com/urfave/cli.(*App).Run
	/home/ubuntu/go/src/chainlink/vendor/github.com/urfave/cli/app.go:255
main.Run
	/home/ubuntu/go/src/chainlink/main.go:61
main.main
	/home/ubuntu/go/src/chainlink/main.go:12
runtime.main
	/usr/local/go/src/runtime/proc.go:195"}

**Comment by dimroc** [comment]
It's unable to connect to Geth's websocket port it seems. Are you sure `8546` is the websocket port and not the JSONRPC port? Let's isolate whether or not it's a `geth` configuration issue. Can you try the following:

1. Navigate to the top level of the repo folder. i.e. `/go/src/chainlink`
2. Ensure docker is installed
3. `./internal/bin/devnet`
4. `./internal/bin/cldev` in another window

If that works, then try the following:
1. `./internal/bin/devnet`
2. `ETH_URL=ws://localhost:18546 ETH_CHAIN_ID=17 chainlink node` in another window

Another option to investigate would be the chain id.

**Comment by dimroc** [comment]
According to your call stack, you are erroring out when creating the ~/.chainlink directory:
https://github.com/smartcontractkit/chainlink/blob/master/store/store.go?utf8=%E2%9C%93#L30-L33
```golang
	err := os.MkdirAll(config.RootDir, os.FileMode(0700))
	if err != nil {
		logger.Fatal(err)
	}
```

Is there a permissions issue with your home directory or the user/system you're running with?

**Comment by thodges-gh** [comment]
Can you give the parameters you used to invoke Geth?
It should be something like this:
```
geth --light --ws --wsaddr 127.0.0.1 --wsport 8546 --wsorigins "*"
```

**Comment by recursiev** [comment]
That's what it was! I didn't know to start geth in websocket mode, that's all there was to it.  Thank you!

**Pull request merged**